pyne/alara.py

```diff
@@ -230,16 +249,33 @@ def photon_source_hdf5_to_mesh(mesh, filename, tags):
             matched_data = h5f.root.data.read_where(
                 "(nuc == '{0}') & (time == '{1}')".format(nuc, cond[1]))
 
-        idx = 0
-        for i, _, ve in mesh:
-            if matched_data[idx][0] == i:
-                tag_handles[tags[cond]][ve] = matched_data[idx][3]
-                idx += 1
-            else:
-                tag_handles[tags[cond]][ve] = [0] * num_e_groups
+        if not sub_voxel:
+            idx = 0
+            for i, _, ve in mesh:
+                if matched_data[idx][0] == i:
+                    tag_handles[tags[cond]][ve] = matched_data[idx][3]
+                    idx += 1
+                else:
+                    tag_handles[tags[cond]][ve] = [0] * num_e_groups
+        else:
+            temp_mesh_data = np.empty(
+                shape=(num_vol_elements, num_e_groups * max_cell_number))
+            for ve in range(num_vol_elements):
+                temp_mesh_data[ve] = [0.0] * num_e_groups * max_cell_number
+            for sve, row in enumerate(subvoxel_array):
+                for i, _, ve in mesh:
```

Comment:
Loop ordering again...  I think `subvoxel_array` is created by looping through the mesh once already, so maybe it can be done more efficiently here

Reply:
I found that the second loop `for i, _, ve in mesh:` is actually not used. So I just delete this second loop.